src/Essentials/src/FileSystem/FileSystem.android.cs

```diff
@@ -68,4 +69,32 @@ internal virtual Task<Stream> PlatformOpenReadAsync()
 			return Task.FromResult<Stream>(stream);
 		}
 	}
+
+	public partial class MediaFileResult
+	{
+		readonly Android.Net.Uri _uri;
+		readonly string _tempFilePath;
+
+		internal MediaFileResult(string fileName, Android.Net.Uri uri, string tempFilePath = null)
+		{
+			_tempFilePath = tempFilePath;
+			_uri = uri;
+			NameWithoutExtension = Path.GetFileNameWithoutExtension(fileName);
+			Extension = Path.GetExtension(fileName).TrimStart('.');
+			ContentType = PlatformGetContentType(Extension);
+			Type = GetFileType(ContentType);
+			FileName = GetFileName(NameWithoutExtension, Extension);
+		}
+
+		internal override Task<Stream> PlatformOpenReadAsync()
+			=> Task.FromResult(Platform.AppContext.ContentResolver!.OpenInputStream(_uri));
+
+		void PlatformDispose()
+		{
+			if (!string.IsNullOrWhiteSpace(_tempFilePath) && File.Exists(_tempFilePath))
+				File.Delete(_tempFilePath);
```

Comment:
There is not need to do a  File.Exists, File.Delete already does that check, from https://learn.microsoft.com/en-us/dotnet/api/system.io.file.delete?view=net-7.0#remarks

> If the file to be deleted does not exist, no exception is thrown.